nearlib/nearclient.js

```diff
@@ -37,22 +37,17 @@ class NearClient {
             args = {};
         }
         const serializedArgs = Buffer.from(JSON.stringify(args)).toString('hex');
-        const result = await this.jsonRpcRequest('abci_query', [`call/${contractAccountId}/${methodName}`, serializedArgs, '0', false]);
-        const response = result.response;
-        let logs = [];
-        if (response.log !== undefined && response.log.length > 0) {
-            logs = response.log.split('\n');
-        }
-        logs.forEach(line => {
-            console.log(`[${contractAccountId}]: ${line}`);
-        });
-        // If error, raise exception after printing logs.
-        const code = response.code || 0;
-        if (code != 0) {
-            throw Error(response.info);
+        const ignoreErrors = true;
+        try {
+            const result = await this.jsonRpcRequest('abci_query', [`call/${contractAccountId}/${methodName}`, serializedArgs, '0', false], ignoreErrors);
+            const response = result.response;
+            _printLogs(contractAccountId, response.log);
+            const json = JSON.parse(_base64ToBuffer(response.value).toString());
+            return json;
+        } catch(e) {
+            _printLogs(contractAccountId, e.log);
+            throw e;
         }
-        const json = JSON.parse(_base64ToBuffer(response.value).toString());
-        return json;
     }
 
     async getTransactionStatus(transactionHash) {
```

Comment:
does this need a doc? I'd like to definitely see some common status codes explained.

Reply:
none of the methods in this class are currently documented. Probably not a good time to add comment, given this is somewhat unrelated bug fix.